Userland/Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -2211,14 +2213,14 @@ Gfx::Font const& StyleComputer::initial_font() const
     return StyleProperties::font_fallback(false, false);
 }
 
-void StyleComputer::absolutize_values(StyleProperties& style) const
+void StyleComputer::absolutize_values(StyleProperties& style, DOM::Element const* element) const
 {
     Length::FontMetrics font_metrics {
-        m_root_element_font_metrics.font_size,
+        element && !is<HTML::HTMLHtmlElement>(*element) ? 16 : m_root_element_font_metrics.font_size,
         style.first_available_computed_font().pixel_metrics()
     };
 
-    auto font_size = style.property(CSS::PropertyID::FontSize)->as_length().length().to_px(viewport_rect(), font_metrics, m_root_element_font_metrics);
+    auto font_size = style.property(CSS::PropertyID::FontSize)->as_length().length().to_px(viewport_rect(), font_metrics, font_metrics);
     font_metrics.font_size = font_size;
```

Comment:
Can you explain what this change is doing? I'm a bit confused, but that could just be me.
- You're only using the root element's font metrics when calculating lengths on the root element.
- Calculating the font size isn't using the root element font metrics at all.

Apparently it doesn't break any tests, but wouldn't that mess up cases where the root element's font size is set to something other than 16?

As a side note, if you do need to hard-code the default font size here, then rather than `16`, it'd be better to use `InitialValues::font_size()` or `m_default_font_metrics.font_size` so that it's only defined in one place. (`m_default_font_metrics` should probably use that `InitialValues` one actually...)

Reply:
I think I mistyped and put the `!` there, it no longer makes sense with it. Regardless, I think it's strange that none of the tests failed with this. I did some further digging and found that the only thing that `absolutize_values` actually does is set the line height. `StyleValue::absolutized` no longer does anything.

Maybe I should close this PR and try to refactor/rewrite the way we do fonts right now (before issues like this are fixed)?

Reply:
`absolutize_values()` should be doing things. Look here:

https://github.com/LadybirdBrowser/ladybird/blob/9fb44cb05777c6d7a8a1950258edadfcee6d4e09/Userland/Libraries/LibWeb/CSS/StyleComputer.cpp#L2313-L2318

It replaces each property's value with an absolutized one. `StyleValue::absolutized()` is a no-op, but it's overridden by several StyleValue types (and should probably be overridden by more of them...) to replace relative length values with absolute ones.